openedx/frontend-app-authoring
# Cleanups for the video editor [FC-0062]

## Description

Currently, the video editor has a ton of propTypes warnings that are logged to the console. This PR fixes some of those warnings by converting the code in question from propTypes (which is deprecated) to TypeScript.

Example warnings are:
```
Warning: Failed prop type: The prop `close` is marked as required in `VideoEditorModal`, but its value is `undefined`.
    at VideoEditorModal (webpack-internal:///./src/editors/containers/VideoEditor/components/VideoEditorModal.jsx:48:5)

Warning: Failed prop type: The prop `isOpen` is marked as required in `VideoEditorModal`, but its value is `undefined`.
    at VideoEditorModal (webpack-internal:///./src/editors/containers/VideoEditor/components/VideoEditorModal.jsx:48:5)

Warning: Failed prop type: Invalid prop `isLibrary` of type `boolean` supplied to `VideoSettingsModal`, expected `function`.
    at VideoSettingsModal (webpack-internal:///./src/editors/containers/VideoEditor/components/VideoSettingsModal/index.jsx:55:5)

```

## Supporting information

Part of https://github.com/openedx/frontend-app-authoring/issues/1092

Private ref: [FAL-3827](https://tasks.opencraft.com/browse/FAL-3827)

## Testing instructions

Open your browser console then edit a video component in a course. Observe fewer warnings. Verify that the editor works exactly the same as it did before.

## Other information

I haven't fixed _all_ the warnings - this is just a start.
